flake.nix

```diff
@@ -0,0 +1,96 @@
+{
+  description = "Qtile's flake, full-featured, hackable tiling window manager written and configured in Python";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-unstable";
+  };
+
+  outputs = { self, nixpkgs }: let
+    supportedSystems = [
+      "x86_64-linux"
+      "aarch64-linux"
+    ];
+
+    forAllSystems = function:
+      nixpkgs.lib.genAttrs supportedSystems
+        (system: function nixpkgs.legacyPackages.${system});
+  in {
+    overlays.default = import ./nix/overlays.nix self;
+
+    checks = forAllSystems (pkgs: pkgs.python3Packages.qtile.passthru.tests);
+
+    packages = forAllSystems (pkgs: let
+      pypkgs = pkgs.python3Packages;
+
+      override-qtile = nixpkgs-qtile:
+        nixpkgs-qtile.overrideAttrs (prev: {
+          name = "${nixpkgs-qtile.pname}-${nixpkgs-qtile.version}";
+          passthru.unwrapped = nixpkgs-qtile;
+        });
+    in {
+      default = self.packages.${pkgs.system}.qtile;
+      qtile = (override-qtile pypkgs.qtile);
+    });
+
+    devShells = forAllSystems (pkgs: let
+      common-shell = {
+        env = {
+          QTILE_DLOPEN_LIBGOBJECT = "${pkgs.glib}/lib/libgobject-2.0.so.0";
```

Comment:
These env vars don't do anything now, for `nix develop` to work you need: https://github.com/jwijenbergh/qtile/commit/e312855cddb20a123a965f40f4b3094995f5d937 (the parts where it patches our dlopen calls)

Hence why I haven't PRed this myself yet as I don't know a good solution to this problem

Reply:
See also that commit I linked how to replace GitHub actions with nix :)

Reply:
Some test fixes from that commit are also good to pull in here

Reply:
we can fix this using
```nix
        env.LD_LIBRARY_PATH = pkgs.lib.makeLibraryPath [
          pkgs.glib
          pkgs.pango
          pkgs.pango
          pkgs.xcb-util-cursor
          pkgs.pixman
          pkgs.libdrm.dev
        ];
```
but patching will be required for cffi build script